test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -103,381 +108,665 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 
-			ginkgo.By("checking that pod is unsuspended when workload is admitted")
-			clusterQueue := testing.MakeClusterQueue("cluster-queue").
-				ResourceGroup(
-					*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "1").Obj(),
-				).Obj()
-			admission := testing.MakeAdmission(clusterQueue.Name).
-				Assignment(corev1.ResourceCPU, "default", "1").
-				AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
-				Obj()
-			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
-			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+				ginkgo.By("checking that pod is unsuspended when workload is admitted")
+				clusterQueue := testing.MakeClusterQueue("cluster-queue").
+					ResourceGroup(
+						*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "1").Obj(),
+					).Obj()
+				admission := testing.MakeAdmission(clusterQueue.Name).
+					Assignment(corev1.ResourceCPU, "default", "1").
+					AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+					Obj()
+				gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
 
-			gomega.Eventually(func() []corev1.PodSchedulingGate {
-				if err := k8sClient.Get(ctx, lookupKey, createdPod); err != nil {
-					return createdPod.Spec.SchedulingGates
-				}
-				return createdPod.Spec.SchedulingGates
-			}, util.Timeout, util.Interval).ShouldNot(
-				gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
-			)
-			gomega.Eventually(func(g gomega.Gomega) bool {
-				ok, err := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
-				g.Expect(err).NotTo(gomega.HaveOccurred())
-				return ok
-			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
-			gomega.Expect(len(createdPod.Spec.NodeSelector)).Should(gomega.Equal(1))
-			gomega.Expect(createdPod.Spec.NodeSelector["kubernetes.io/arch"]).Should(gomega.Equal("arm64"))
-
-			gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
-			gomega.Expect(createdWorkload.Status.Conditions).Should(gomega.BeComparableTo(
-				[]metav1.Condition{
-					{Type: kueue.WorkloadQuotaReserved, Status: metav1.ConditionTrue},
-					{Type: kueue.WorkloadAdmitted, Status: metav1.ConditionTrue},
-				},
-				wlConditionCmpOpts...,
-			))
+				util.ExpectPodUnsuspendedWithNodeSelectors(ctx, k8sClient, lookupKey, map[string]string{"kubernetes.io/arch": "arm64"})
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					ok, err := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
+					g.Expect(err).To(gomega.Succeed())
```

Comment:
It's the same, but this reads better

```suggestion
					g.Expect(err).NotTo(gomega.HaveOcurred())
```